dali/base/dadfs.cpp

```diff
@@ -7608,9 +7621,12 @@ IDistributedFile *CDistributedFileDirectory::dolookup(CDfsLogicalFileName &_logi
     return NULL;
 }
 
-IDistributedFile *CDistributedFileDirectory::lookup(CDfsLogicalFileName &logicalname, IUserDescriptor *user, bool writeattr, bool hold, bool lockSuperOwner, IDistributedFileTransaction *transaction, unsigned timeout)
+IDistributedFile *CDistributedFileDirectory::lookup(CDfsLogicalFileName &logicalname, IUserDescriptor *user, bool writeattr, bool hold, bool lockSuperOwner, IDistributedFileTransaction *transaction, bool privilegedUser, unsigned timeout)
 {
-    return dolookup(logicalname, user, writeattr, hold, lockSuperOwner, transaction, timeout);
+    Owned <IDistributedFile> distributedFile = dolookup(logicalname, user, writeattr, hold, lockSuperOwner, transaction, timeout);
+    if (writeattr==false && distributedFile && distributedFile->isRestrictedAccess() && !privilegedUser)
+        throw new CDFS_Exception(DFSERR_RestrictedFileAccessDenied,logicalname.get());
```

Comment:
I think it's deliberate, but writing to a file that is restricted by a non-privileged user - is allowed, correct?

If so, I think should add a comment here though to explain.

Reply:
Comment added